### ISSUE TITLE
[sugestion] Get more precise start date without log module

### ISSUE THREAD

**Issue by NuxitBrain** [echemdata/galvani#94]
dear all ! 

For the moment, the self.startdate only give the start date in day. For some application, we need more precise launch time ( at least hour base). I now that we can get more precise timing with self.timestamp but for that, we need to end the programe ( creation of the log part). Do you thing it possible to read the the timestamp values at another place in the raw file ?  or if the start date is whriten somewhere else more precisely ? 

thank you in advance

**Comment by jojo-Li2** [comment]
Yes it seems that the timestamp is exclusively in the log module. And the log module is added to the .mpr file when the experiment is ended, before that it is stored in a .mpl log file.
So I think the easiest way would be to check if the .mpl is present in the folder where the .mpr is stored, and extract the timestamp with basic regular expression handling.

This could also be done during initialization of the MPRfile class. Same for the loop module.

@PeterKraus what is the external device module ? I couldn't find it

**Comment by PeterKraus** [comment]
I think some potentiostats allow you to add an external sensor, such as a temperature and/or pressure sensor (really just a calibrated voltage readout). For the `dtypes` we parse in `yadg`, see [here](https://github.com/dgbowl/yadg/blob/1b2c7f8d1324a7b2b3eb99d1d430351db40bcd08/src/yadg/parsers/electrochem/eclabcommon/mpr_columns.py#L182).
